turtlebot2_drivers/src/kobuki_node.cpp

```diff
@@ -60,12 +60,27 @@ int main(int argc, char * argv[])
 {
   rclcpp::init(argc, argv);
 
+  // TODO(clalancette): we set the depth to 50 here since it seems to help workaround
+  // a bug where rclcpp::spin_some() can go into an infinite loop sometimes.  We should
+  // find and fix the root cause instead of this workaround.
+  rmw_qos_profile_t cmd_vel_qos_profile;
+  cmd_vel_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  cmd_vel_qos_profile.depth = 50;
+  cmd_vel_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  cmd_vel_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
+
+  rmw_qos_profile_t sensor_qos_profile;
+  sensor_qos_profile.history = RMW_QOS_POLICY_HISTORY_KEEP_LAST;
+  sensor_qos_profile.depth = 50;
+  sensor_qos_profile.reliability = RMW_QOS_POLICY_RELIABILITY_BEST_EFFORT;
+  sensor_qos_profile.durability = RMW_QOS_POLICY_DURABILITY_VOLATILE;
+
   auto node = rclcpp::node::Node::make_shared("kobuki_node");
   auto parameter_service = std::make_shared<rclcpp::parameter_service::ParameterService>(node);
   auto cmd_vel_sub = node->create_subscription<geometry_msgs::msg::Twist>(
-    "cmd_vel", cmdVelCallback, rmw_qos_profile_default);
-  auto odom_pub = node->create_publisher<nav_msgs::msg::Odometry>("odom", rmw_qos_profile_default);
-  auto imu_pub = node->create_publisher<sensor_msgs::msg::Imu>("imu", rmw_qos_profile_default);
+    "cmd_vel", cmdVelCallback, cmd_vel_qos_profile);
+  auto odom_pub = node->create_publisher<nav_msgs::msg::Odometry>("odom", sensor_qos_profile);
+  auto imu_pub = node->create_publisher<sensor_msgs::msg::Imu>("imu", sensor_qos_profile);
```

Comment:
Ok, I'm not against keeping these settings if we are not sure if changing them would break things. Can you rename to a name more differentiablefrom our defined `rmw_qos_profile_sensor_data` ?